image_classification/utils/utils.py

```diff
@@ -0,0 +1,121 @@
+    elif experiment == 'traditional-kd' and hyper_params['stage'] == 0:
+        for name, param in model.named_parameters() : 
+            param.requires_grad = False
+            if (name[0] == 'b' or name[0] == 'c' or name[5] == str(0) or name[5] == str(1) or name[5] == str(2)) : 
+                param.requires_grad = True
+    # traditional-kd last stage
+    elif experiment == 'traditional-kd' and hyper_params['stage'] == 1:
+        for name, param in model.named_parameters() : 
+            param.requires_grad = False
+            if not (name[0] == 'b' or name[0] == 'c' or name[5] == str(0) or name[5] == str(1) or name[5] == str(2)) : 
+                param.requires_grad = True
+
+    return model
+
+
+def get_savename(hyper_params, experiment):
+    assert experiment in ['stagewise-kd', 'traditional-kd', 'simultaneous-kd', 'no-teacher']
+    less = 'full_data'
+    if hyper_params['percentage'] is not None:
+        less = 'less_data' + str(hyper_params['percentage'])
+
+    if experiment == 'stagewise-kd':
+        savename = '../saved_models/' + str(hyper_params['dataset']) + '/' + less + '/stagewise-kd/' + str(hyper_params['model']) + '_stage' + str(hyper_params['stage'])
+    
+    elif experiment == 'traditional-kd':
+        savename = '../saved_models/' + str(hyper_params['dataset']) + '/' + less + '/traditional-kd/' + str(hyper_params['model']) + '_stage' + str(hyper_params['stage'])
+
+    elif experiment == 'simultaneous-kd':
+        savename = '../saved_models/' + str(hyper_params['dataset']) + '/' + less + '/simultaneous-kd/' + str(hyper_params['model'])
+        
+    elif experiment == 'no-teacher':
+        savename = '../saved_models/' + str(hyper_params['dataset']) + '/' + less + '/no-teacher/' + str(hyper_params['model']) + '_classifier'
+    
+    os.makedirs(savename, exist_ok=True)
+    return savename + '/model' + str(hyper_params['seed']) + '.pt'
+
+
+def get_model(model_name, dataset, data=None, teach=False):
+    if teach and data:
+        load_name = dataset
+        if dataset == 'cifar10' : 
+            load_name = dataset[ : -2]
+        else:
+            load_name = dataset + '2'
+        teacher = cnn_learner(data, models.resnet34, metrics=accuracy, pretrained=False)
+        teacher = teacher.load(os.path.expanduser("~") + '/.fastai/data/' + load_name + '/models/resnet34_' + load_name + '_bs64')
+        teacher.freeze()
+
+    if model_name == 'resnet10' :
+        net = resnet10(pretrained=False, progress=False)
+    elif model_name == 'resnet14' : 
+        net = resnet14(pretrained=False, progress=False)
+    elif model_name == 'resnet18' :
+        net = resnet18(pretrained=False, progress=False)
+    elif model_name == 'resnet20' :
+        net = resnet20(pretrained=False, progress=False)
+    elif model_name == 'resnet26' :
+        net = resnet26(pretrained=False, progress=False)
+    elif model_name == 'resnet34' :
+        net = resnet34(pretrained=False, progress=False)
```

Comment:
since all of them have the same parameters you should use `getattr` function.
``` 
net = getattr(model_name)(pretrained=False, progress=False)
```

Reply:
yes, I was looking at this. Wasn't sure how it worked. I'll try it.

Reply:
It should be:
getattr(model_name, custom_resnet)

Reply:
done. It's actually `getattr(custom_resnet, model_name)`